src/legacy/core_plugins/kibana/public/dashboard/saved_dashboard/saved_dashboard.d.ts

```diff
@@ -25,7 +25,7 @@ import { Query } from 'src/legacy/core_plugins/data/public';
 import { Filter } from '@kbn/es-query';
 
 export interface SavedObjectDashboard extends SavedObject {
-  id?: string;
+  id: string;
```

Comment:
Did you change this because of the `map()` in the list view? I think it is possible that there is a `SavedObjectDashboard` around without an id (in cases where the dashboard isn't saved yet), so I would like to not change the type here. You can filter out the dashboards without id to stop typescript from complaining (it won't be a runtime problem because there won't be unsaved dashboards in the listing even though that might change in the future):
```ts
selectedIds: obj.map(item => item.id).filter((id: undefined | string): id is string => Boolean(id))
```